src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -111,33 +100,12 @@ private void Workspace_WorkspaceChanged(object? sender, WorkspaceChangeEventArgs
                     EnqueueUpdate(e.NewSolution.GetProject(projectId));
                 }
 
-                break;
-            case WorkspaceChangeKind.SolutionCleared:
-            case WorkspaceChangeKind.SolutionRemoved:
-                foreach (var project in e.OldSolution.Projects)
-                {
-                    RemoveProject(project);
-                }
-
                 break;
             default:
                 break;
         }
     }
 
-    private void RemoveProject(Project? project)
```

Comment:
This never handled removing the file, it just stopped the queue. I'm not actually sure what handles cleanup. Since we're working on `_workspace.CurrentSolution` I'm guessing it would be safe to just delete the file when a project is removed?

That's new, and maybe it's already done somewhere else? Not sure

Reply:
One thought: 

We could enqueue and update for the removed `ProjectId` and in the ABWQ work method we could clean up the file if it exists and the project is no longer in the solution?

Reply:
This was probably unnecessary originally, but yeah I was just trying to remove the scheduler. We'd never get any more updates for the project anyway, so could have ignored removes then too.

Not sure I like the idea of removing the file though. I think in VS Code we are more likely to be processing requests before the file is written, so it does serve more like an offline cache, even if it is one that won't be being updated any more.

Reply:
Yeah, I agree that this is probably OK. I saw all of the red and notice that all it did was get rid of the scheduler before. With #10475 I think VS Code will start reading in `.bin` even sooner, so you're right that VS Code will use it more as a cache.

Reply:
Test failures lead to one thing remove did that I didn't notice: we only queue work for items that have been reported with `NotifyDynamicFile`. This PR breaks that. I'm adding remove back to fix